### PR TITLE
cleanup pre-commit.sh

### PR DESCRIPTION
**Pull request by percyliang**
### Reasons for making this change

On a fresh checkout of master, `./pre-commit.sh` was failing.  I removed the `--check` flag, and it works now.

EDIT: added `--check` back in (turns out I was using the wrong npm version).  So this PR basically just cleans up the `pre-commit.sh` script.

### CODE REVIEW

File: codalab/lib/spec_util.py [hunk 1]
```diff
@@ -11,7 +11,9 @@
 UUID_PREFIX_REGEX = re.compile('^0x[0-9a-f]{1,31}$')
 BEGIN_NAME_STR = '[a-zA-Z_]'
 NAME_STR = '[a-zA-Z_][a-zA-Z0-9_\.\-]*'
-NAME_PATTERN_STR = '[%\*a-zA-Z0-9_\.\-]+'  # Allow % for matching wildcard (SQL syntax), and * (regular expressions)
+NAME_PATTERN_STR = (
```
Comment (percyliang) [inline codalab/lib/spec_util.py]: re-format

File: frontend/package.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "check-ci": "CI=true eslint src && prettier --config ./.prettierrc --check 'src/**/*.js'",
-    "format": "CI=true eslint src --fix && prettier --config ./.prettierrc --check 'src/**/*.js' --write"
+    "format": "CI=true eslint src --fix && prettier --config ./.prettierrc 'src/**/*.js' --write"
```
Comment (percyliang) [inline frontend/package.json]: key change

File: rc [hunk 3]
```diff
@@ -31,3 +31,6 @@ if [ ! -e "$codalabdir/venv/bin/register-python-argcomplete" ]; then
 else
     eval "$($codalabdir/venv/bin/register-python-argcomplete cl)"
 fi
+
+# Used the version of `cl` that's installed into `venv` by default.
+. venv/bin/activate
```
Comment (percyliang) [inline rc:36]: so if you `. rc` and run `pre-commit.sh`, then you can use the latest version for quick testing.

File: pre-commit.sh [hunk 4]
```diff
@@ -18,5 +18,5 @@ venv/bin/mkdocs build || { rm -rf venv; exit 1; }  # Outputs to `site`
 # Note: run `venv/bin/mkdocs serve` for a live preview
 
 # Fix Python and JavaScript style (mutates code!)
-venv/bin/black codalab scripts *.py || exit
-npm run --prefix frontend format
+venv/bin/black codalab scripts *.py || exit 1
+npm run --prefix frontend format || exit 1
```
Comment (percyliang) [inline pre-commit.sh]: Drive-by cleanup

Comment (epicfaace) [reply pre-commit.sh]: Can we do `set -e` at the beginning of the script and remove the `exit 1`'s everywhere?

File: frontend/package.json (hunk 2 above)
Comment (epicfaace) [reply frontend/package.json]: Why remove `--check`?

File: pre-commit.sh (hunk 4 above)
Comment (percyliang) [reply pre-commit.sh]: good idea, done

File: frontend/package.json (hunk 2 above)
Comment (percyliang) [reply frontend/package.json]: Reverted...turned out I had the wrong version of prettier.

**Review by epicfaace — APPROVED** [review]
